### PR TITLE
Move subp into its own module.

### PR DESCRIPTION
**Pull request by smoser**
This was painful, but it finishes a TODO from cloudinit/subp.py.

It moves the following from util to subp:
  ProcessExecutionError
  subp_blob_in_tempfile
  subp
  which
  target_path

It is arguable that 'target_path' could be moved to a 'path_utils' or
something, but in order to use it from subp and also from utils,
we had to get it out of utils.

### CODE REVIEW

**Comment by igalic** [comment]
looking at this diff makes me think we should extract the service handling code into its own set of functions


ooooooor, move them into distro 😅

File: cloudinit/subp.py [hunk 1]
```diff
@@ -54,4 +56,299 @@ def prepend_base_command(base_command, commands):
...
+
+    for path in search:
+        ppath = os.path.sep.join((path, program))
+        if is_exe(target_path(target, ppath)):
+            return ppath
+
+    return None
+
+
+def is_exe(fpath):
```
Comment (blackboxsw) [inline cloudinit/subp.py:349]: Can we make this private since it's only a helper function used locally which is not intended for external consumers?

**Comment by smoser** [comment]
> Looks good, three somewhat minor notes.
> 
>     1. is_exe -> _is_exe
> 
>     2. Can we move util.runparts -> subp.runparts
> 
>     3. subp.runparts  to call _is_exe too

I think I agree with all of these, but suggest that maybe they are all sane as follow-on.

Thoughts?

**Comment by smoser** [comment]
> > Looks good, three somewhat minor notes.
> > ```
> > 1. is_exe -> _is_exe

OK, I went ahead and did this one.

**Comment by blackboxsw** [comment]
> > Looks good, three somewhat minor notes.
> > ```
> > 1. is_exe -> _is_exe
> > 
> > 2. Can we move util.runparts -> subp.runparts
> > 
> > 3. subp.runparts  to call _is_exe too
> > ```
> 
> I think I agree with all of these, but suggest that maybe they are all sane as follow-on.
> 
> Thoughts?

Yes let's pull in parts 2 & 3 as a separate PR.

**Comment by smoser** [comment]
2 -> #420 
3 -> #419

**Pull request merged**